src/publisher_service/worker.rs

```diff
@@ -0,0 +1,258 @@
+                WHERE id = notification
+            ) AS project
+    ";
+    sqlx::query_as::<Postgres, ProcessingMessageItem>(pick_message_query)
+        .fetch_optional(pg_pool)
+        .await
+}
+
+/// Creates a message to be sent to the relay and calling the relay client
+#[instrument]
+async fn process_message(
+    project_id: &str,
+    notification_id: &str,
+    account_id: &str,
+    pg_pool: &PgPool,
+    relay_client: Arc<relay_client::http::Client>,
+) -> Result<(), anyhow::Error> {
+    let project = get_project_by_project_id(project_id.into(), pg_pool).await?;
+    let project_signing_details = {
+        let private_key = ed25519_dalek::SigningKey::from_bytes(&decode_key(
+            &project.authentication_private_key,
+        )?);
+        let decoded_client_id = DecodedClientId(
+            hex::decode(project.authentication_public_key.clone())?[0..32].try_into()?,
+        );
+        let identity = ClientId::from(decoded_client_id);
+        ProjectSigningDetails {
+            identity,
+            private_key,
+            app: project.app_domain.into(),
+        }
+    };
+
+    let id = chrono::Utc::now().timestamp_millis().unsigned_abs();
+    let notification = get_notification_by_notification_id(notification_id, pg_pool).await?;
+    let message = JsonRpcPayload {
+        id,
+        jsonrpc: "2.0".to_string(),
+        params: JsonRpcParams::Push(NotifyPayload {
+            message_auth: sign_message(notification, account_id.into(), &project_signing_details)?
+                .to_string(),
+        }),
+    };
+
+    let subscriber = get_subscriber_by_subscriber_id(account_id, pg_pool).await?;
+    let sym_key = decode_key(&subscriber.sym_key)?;
+    let envelope = Envelope::<EnvelopeType0>::new(&sym_key, &message)?;
+    let base64_notification = base64::engine::general_purpose::STANDARD.encode(envelope.to_bytes());
+    let topic = Topic::new(sha256::digest(&sym_key).into());
+
+    let publish = Publish {
+        topic,
+        message: base64_notification.into(),
+        ttl_secs: NOTIFY_MESSAGE_TTL.as_secs() as u32,
+        tag: NOTIFY_MESSAGE_TAG,
+        prompt: true,
+    };
+    do_publish(relay_client.clone(), account_id, &publish)
+        .await
+        .unwrap();
+    Ok(())
+}
+
+/// Publishes a notification to the relay using the relay client
+#[instrument]
+async fn do_publish(
+    relay_client: Arc<relay_client::http::Client>,
+    account_id: &str,
+    publish: &Publish,
+) -> Result<(), relay_client::error::Error> {
+    let client_publish_call = || {
+        relay_client.publish(
+            publish.topic.clone(),
+            publish.message.clone(),
+            publish.tag,
+            Duration::from_secs(publish.ttl_secs as u64),
+            publish.prompt,
+        )
+    };
+
+    // Handling retrying attempts on temporary relay errors
+    let mut tries = 0;
+    while let Err(e) = client_publish_call().await {
+        tries += 1;
+        if tries >= 10 {
```

Comment:
My thinking was that we should retry once or twice here, to cover known temporary failures. But if it fails (e.g. relay down) it should still be put back into the queue to be retried again later.